src/lightly_train/_task_models/dinov2_eomt_semantic_segmentation/task_model.py

```diff
@@ -211,16 +210,19 @@ def forward(self, x: Tensor) -> tuple[Tensor, Tensor]:
         return masks, logits
 
     # TODO(Guarin, 07/25): Refactor to take attn_mask_probs as input.
-    def forward_train(self, x: Tensor) -> tuple[list[Tensor], list[Tensor]]:
-        B, C, H, W = x.shape
+    def forward_train(
+        self, x: Tensor, return_logits_per_layer: bool
+    ) -> tuple[list[Tensor], list[Tensor]]:
+        _, _, H, W = x.shape
         patch_size = self.backbone.patch_size
         grid_size = (H // patch_size, W // patch_size)
 
         x = self.backbone.prepare_tokens_with_masks(x)  # type: ignore[no-untyped-call]
-        attn_mask = None
         mask_logits_per_layer, class_logits_per_layer = [], []
 
         for i, block in enumerate(self.backbone.blocks):
+            attn_mask = None
```

Comment:
Moved this inside the loop to make it more evident that a new attention mask is created in each iteration.